google/cloud/aiplatform/initializer.py

```diff
@@ -149,6 +149,35 @@ def project(self) -> str:
         if self._project:
             return self._project
 
+        # Project is not set. Trying to get it from the environment.
+        # See https://github.com/googleapis/python-aiplatform/issues/852
+        # See https://github.com/googleapis/google-auth-library-python/issues/924
+        # TODO: Remove when google.auth.default() learns the
+        # CLOUD_ML_PROJECT_ID env variable or Vertex AI starts setting GOOGLE_CLOUD_PROJECT env variable.
+        project_number = os.environ.get("CLOUD_ML_PROJECT_ID")
+        if project_number:
+            self._project = project_number
+            # Try to convert project number to project ID which is more readable.
+            try:
+                from googleapiclient import discovery
```

Comment:
Preference to move the conversion of project number to project ID to shared utility so it can used elsewhere in the future. This could be useful in scenarios where project number is not supported, like BQ.

cc: @morgandu

Reply:
Done.